test/plugin.test.js

```diff
@@ -354,24 +356,34 @@ test('check dependencies - should throw', t => {
 })
 
 test('set the plugin name based on the plugin displayName symbol', t => {
-  t.plan(6)
+  t.plan(16)
   const fastify = Fastify()
 
   fastify.register(fp((fastify, opts, done) => {
     t.equal(fastify.pluginName, 'fastify -> plugin-A')
+    t.ok(fastify.hasPlugin('plugin-A'))
     fastify.register(fp((fastify, opts, done) => {
       t.equal(fastify.pluginName, 'fastify -> plugin-A -> plugin-AB')
+      t.ok(fastify.hasPlugin('plugin-A'))
+      t.ok(fastify.hasPlugin('plugin-AB'))
```

Comment:
This is an interesting case. What would happen if the chain includes "plugin-AB" but _not_ "plugin-A" and we do `hasPlugin('plugin-A')`?

Reply:
@jsumners 
Thanks. I added test cases. Could you review it again? 🙏 
https://github.com/fastify/fastify/pull/3963/commits/0a3b4fd467577f25f437434705e4c6215e46265a

Reply:
@jsumners 
I see. Thanks for the explanation.
I guess it would return `false`. Because `pluginName` joins the array data that is stored in `this[kPluginNameChain]` and `hasPlugin` simply searches the name in the array. So if the name is not stored in `pluginName` result, `hasPlugin` would return `false`.